src/main/java/it/gov/pagopa/payhub/auth/service/user/IamUserInfoDTO2UserInfoMapper.java

```diff
@@ -9,77 +11,111 @@
-		if (iamUserInfoDTO.isSystemUser()) {
-			return systemUserMapper(iamUserInfoDTO);
-		}
-		return userInfoMapper(iamUserInfoDTO);
-	}
-
-	private UserInfo systemUserMapper(IamUserInfoDTO iamUserInfoDTO) {
-		String organizationIpaCode = iamUserInfoDTO.getOrganizationAccess().getOrganizationIpaCode();
-		return UserInfo.builder()
-			.userId(iamUserInfoDTO.getUserId())
-			.mappedExternalUserId(buildSystemMappedExternalUserId(organizationIpaCode))
-			.fiscalCode(iamUserInfoDTO.getFiscalCode())
-			.familyName(iamUserInfoDTO.getFamilyName())
-			.name(iamUserInfoDTO.getName())
-			.issuer(iamUserInfoDTO.getIssuer())
-			.organizations(Collections.singletonList(UserOrganizationRoles.builder()
-				.organizationIpaCode(organizationIpaCode)
-				.roles(Collections.singletonList(Constants.ROLE_ADMIN))
-				.build()))
-			.build();
-	}
-
-	public static String buildSystemMappedExternalUserId(String organizationIpaCode) {
-		return organizationIpaCode + WS_USER_SUFFIX;
-	}
-
-	private UserInfo userInfoMapper(IamUserInfoDTO iamUserInfoDTO) {
-		User user = usersRepository.findById(iamUserInfoDTO.getInnerUserId()).orElseThrow(() -> new UserNotFoundException("Cannot found user having inner id:" + iamUserInfoDTO.getInnerUserId()));
-		List<Operator> userRoles = operatorsRepository.findAllByUserId(iamUserInfoDTO.getInnerUserId());
-		UserInfo userInfo = UserInfo.builder()
-			.userId(user.getUserId())
-			.mappedExternalUserId(user.getMappedExternalUserId())
-			.fiscalCode(iamUserInfoDTO.getFiscalCode())
-			.familyName(iamUserInfoDTO.getFamilyName())
-			.name(iamUserInfoDTO.getName())
-			.issuer(iamUserInfoDTO.getIssuer())
-			.organizations(userRoles.stream()
-				.map(r -> UserOrganizationRoles.builder()
-					.operatorId(r.getOperatorId())
-					.organizationIpaCode(r.getOrganizationIpaCode())
-					.roles(new ArrayList<>(r.getRoles()))
-					.email(r.getEmail())
-					.build())
-				.toList())
-			.build();
-
-		if(iamUserInfoDTO.getOrganizationAccess() != null){
-			userInfo.setOrganizationAccess(iamUserInfoDTO.getOrganizationAccess().getOrganizationIpaCode());
-		}
-		return userInfo;
-	}
+    public static final String WS_USER_SUFFIX = "-WS_USER";
+    private final UsersRepository usersRepository;
+    private final OperatorsRepository operatorsRepository;
+    private final OrganizationSearchClient organizationSearchClient;
+    @Value("${app.enable-access-organization-mode}")
+    private boolean organizationAccessMode;
+
+    public IamUserInfoDTO2UserInfoMapper(UsersRepository usersRepository, OperatorsRepository operatorsRepository, OrganizationSearchClient organizationSearchClient) {
+        this.usersRepository = usersRepository;
+        this.operatorsRepository = operatorsRepository;
+        this.organizationSearchClient = organizationSearchClient;
+    }
+
+    @Override
+    public UserInfo apply(IamUserInfoDTO iamUserInfoDTO) {
+        throw new UnsupportedOperationException("AccessToken is required for this operation.");
+    }
```

Comment:
why have you introduced this? why don't just remove it and fixing all the logic which is bringing us here?